Core/Lib/Models/MetadataModel.cs

```diff
@@ -10,5 +10,21 @@ public class MetadataModel : IMetadataModel
 
     private Dictionary<string, JsonElement>? metadataDictionary;
     [JsonIgnore]
-    public Dictionary<string, JsonElement>? MetadataDictionary => metadataDictionary ??= JsonSerializer.Deserialize<Dictionary<string, JsonElement>>(Metadata?.ToString() ?? JsonNodeUtilities.EmptyObjectJson);
+    public Dictionary<string, JsonElement>? MetadataDictionary
+    {
+        get
+        {
+            return metadataDictionary;
```

Comment:
Tip: Better to distinguish a bit more using a '_' as prefix or 'Bf'/ 'BackingField' as suffix